atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -448,10 +448,11 @@ private static LockAndTimestampServices withRefreshingLockService(
 
     private static LockAndTimestampServices withRequestBatchingTimestampService(
             java.util.function.Supplier<TimestampClientConfig> timestampClientConfigSupplier,
-            LockAndTimestampServices lockAndTimestampServices) {
-        TimelockService timelockServiceWithBatching =
+            LockAndTimestampServices lockAndTimestampServices, String userAgent) {
+        TimelockService timelockServiceWithBatching = ServiceCreator.createInstrumentedService(
                 DecoratedTimelockServices.createTimelockServiceWithTimestampBatching(
-                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier);
+                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier),
+                TimelockService.class, userAgent);
```

Comment:
I believe we already instrument the `TimelockService` below (line 497), so not sure how the metrics will handle two `TimelockService` interfaces. @schlosna is this ok?